src/js/components/pages/greendash/GreenRecommendation.tsx

```diff
@@ -193,40 +205,26 @@ const PublisherListRecommendations = (): JSX.Element | null => {
 	const pvChartTotalValue = baseFilterConfirmed.prob && baseFilterConfirmed.prob != 0 ? pvChartTotal.value?.sampling : pvChartTotal.value;
 	const bucketsPer1000 = emissionsPerImpressions(pvChartTotalValue.by_domain.buckets);
 
-	let leftDomains = lowBuckets && lowBuckets!.map((val) => val.key as string);
-	let rightDomains = highBuckets && highBuckets!.map((val) => val.key as string);
-	// what weight of impressions is included?
-	let lowImps = 0,
-		highImps = 0,
-		lowWeightedAvg = 0,
-		highWeightedAvg = 0;
-	if (lowBuckets && highBuckets) {
-		lowBuckets.forEach((bucket) => {
-			lowImps += bucket.count as number;
-			lowWeightedAvg += (bucket.count as number) * (bucket.co2 as number);
-		});
-		highBuckets.forEach((bucket) => {
-			highImps += bucket.count as number;
-			highWeightedAvg += (bucket.count as number) * (bucket.co2 as number);
-		});
-		lowWeightedAvg = lowWeightedAvg / lowImps;
-		highWeightedAvg = highWeightedAvg / highImps;
-	}
-	let totalImps = 0,
-		weightedAvg = 0;
-	bucketsPer1000.map((bucket) => {
-		weightedAvg += (bucket.count as number) * (bucket.co2 as number);
-		totalImps += bucket.count as number;
-	});
-	weightedAvg = weightedAvg / totalImps;
-
 	const co2s = bucketsPer1000.map((row) => row.co2! as number);
+	const totalCounts = bucketsPer1000.reduce((acc, cur) => acc + (cur.count as number), 0);
 	const maxCo2 = Math.max(...co2s);
 	const minCo2 = Math.min(...co2s);
 	const middleCo2 = ((maxCo2 - minCo2) * 1) / 2;
 	const steps = (maxCo2 - minCo2) / TICKS_NUM; // How large is a tick
 	const silderProps: RangeSliderProps = { min: minCo2 * 1, max: maxCo2 * 1, step: steps, defaultValue: middleCo2, onChange: setSelectedCo2 };
 
+	// what weight of impressions is included?
+	let lowImps = 0,
+		lowWeightedAvg = 0;
+	lowBuckets?.forEach((bucket) => {
+		lowImps += bucket.count as number;
+		lowWeightedAvg += (bucket.count as number) * (bucket.co2 as number);
+	});
+	lowWeightedAvg = lowWeightedAvg / lowImps;
+	const weightedAvg =
+		bucketsPer1000.reduce((acc, bucket) => acc + (bucket.count as number) * (bucket.co2 as number), 0) /
+		bucketsPer1000.reduce((acc, bucket) => acc + (bucket.count as number), 0);
```

Comment:
Hello @wongws11 
I think you can replace this `bucketsPer1000.reduce((acc, bucket) => acc + (bucket.count as number), 0)` with the `totalCounts` value calculated a few lines above. Which should make the formula shorter and easier to read.

Reply:
You're right. Thanks for reminding me.